charts/osm/values.yaml

```diff
@@ -7,12 +7,25 @@ OpenServiceMesh:
   #
   # -- OSM control plane image parameters
   image:
-    # -- Container image registry
+    # -- Container image registry for control plane images
     registry: openservicemesh
-    # -- Container image pull policy
+    # -- Container image pull policy for control plane containers
     pullPolicy: IfNotPresent
-    # -- Container image tag
-    tag: v0.9.2
+    # -- Container image tag for control plane images
+    tag: "latest-main"
```

Comment:
Let's change the pull policy to `Always` by default if `latest-main` images are used by default.

Reply:
I didn't change this just because the demo scripts sets this to Always. We have to be careful in not setting this to Always for release versions, so it makes sense to default it to `IfNotPresent`.

Reply:
@nojnhuh, what's the reason behind your request to change to `Always`?  It seems to me that `IfNotPresent` is the safest bet for production environments (so Kubernetes does not do any extra unnecessary work) and `Always` perhaps for development.

Reply:
> I didn't change this just because the demo scripts sets this to Always. We have to be careful in not setting this to Always for release versions, so it makes sense to default it to `IfNotPresent`.

I suppose we could effectively get this behavior "for free" if the `latest-main` images were instead named `latest` and take advantage of the [default behavior](https://kubernetes.io/docs/concepts/containers/images/#imagepullpolicy-defaulting) by blanking the pull policy. Not sure how many people are currently relying on `latest` images though.

> what's the reason behind your request to change to Always?

`latest-main` is a floating tag whose meaning changes frequently. If I were running a cluster and installed the chart using `latest-main` images pulled `IfNotPresent`, then installing a new version of the chart some time later without changing the tag or pull policy will still use the old images since images with the `latest-main` tag already exist on the node. The only way to get the new images to pull would be to manually delete them from the node. If the images were pulled `Always`, then each install of OSM would get the newest images automatically. I find those kinds of issues where I'm not running the versions of images that I think I am particularly frustrating.

> It seems to me that `IfNotPresent` is the safest bet for production environments (so Kubernetes does not do any extra unnecessary work) and `Always` perhaps for development.

Isn't that exactly the distinction we're making here with the tags, where the main branch is set up for development and the release branches are set up for production?

Reply:
@nojnhuh I don't mind changing the PullPolicy to `Always` for the `main` branch, but we have to make it a part of the release checklist to ensure the PullPolicy is always set to `IfNotPresent` for released images.
I don't think this should be a major concern for this PR in any way, given that we can easily override it.